cpu/sources/layout7/cpu_oods.move

```diff
@@ -0,0 +1,940 @@
+                push_back(expmods_and_points, point);
+
+                // point *= g^4.
+                point = fmul(point, tg4);
+                // expmods_and_points.points[87] = -(g^1008 * z).
+                push_back(expmods_and_points, point);
+
+                // point *= g^13.
+                point = fmul(point, tg13);
+                // expmods_and_points.points[88] = -(g^1021 * z).
+                push_back(expmods_and_points, point);
+
+                // point *= g.
+                point = fmul(point, trace_generator);
+                // expmods_and_points.points[89] = -(g^1022 * z).
+                push_back(expmods_and_points, point);
+
+                // point *= g.
+                point = fmul(point, trace_generator);
+                // expmods_and_points.points[90] = -(g^1023 * z).
+                push_back(expmods_and_points, point);
+
+                // point *= g.
+                point = fmul(point, trace_generator);
+                // expmods_and_points.points[91] = -(g^1024 * z).
+                push_back(expmods_and_points, point);
+
+                // point *= g.
+                point = fmul(point, trace_generator);
+                // expmods_and_points.points[92] = -(g^1025 * z).
+                push_back(expmods_and_points, point);
+
+                // point *= g^2.
+                point = fmul(point, tg2);
+                // expmods_and_points.points[93] = -(g^1027 * z).
+                push_back(expmods_and_points, point);
+
+                // point *= g^7.
+                point = fmul(point, tg7);
+                // expmods_and_points.points[94] = -(g^1034 * z).
+                push_back(expmods_and_points, point);
+
+                // point *= g.
+                point = fmul(point, trace_generator);
+                // expmods_and_points.points[95] = -(g^1035 * z).
+                push_back(expmods_and_points, point);
+
+                // point *= g^1010.
+                point = fmul(point, tg1010);
+                // expmods_and_points.points[96] = -(g^2045 * z).
+                push_back(expmods_and_points, point);
+
+                // point *= g^13.
+                point = fmul(point, tg13);
+                // expmods_and_points.points[97] = -(g^2058 * z).
+                push_back(expmods_and_points, point);
+            };
+
+            let eval_points_ptr = /*oodseval_points*/ MM_OODS_EVAL_POINTS;
+            let eval_points_end_ptr = eval_points_ptr + (*borrow(ctx, MM_N_UNIQUE_QUERIES) as u64);
+
+            // The batchInverseArray is split into two halves.
+            // The first half is used for cumulative products and the second half for values to invert.
+            // Consequently the products and values are half the array size apart.
+            let products_ptr = 0;
+            // Compute an offset in bytes to the middle of the array.
+            let products_to_values_offset = length(batch_inverse_array) >> 1;
```

Comment:
```suggestion
            let products_to_values_offset = n_queries * BATCH_INVERSE_CHUNK);
```

Reply:
there is no `n_queries` in this function